src/ess/bifrost/io/nexus.py

```diff
@@ -0,0 +1,48 @@
+# SPDX-License-Identifier: BSD-3-Clause
+# Copyright (c) 2024 Scipp contributors (https://github.com/scipp)
+
+"""NeXus input/output for BIFROST."""
+
+import sciline
+import scipp as sc
+import scippnexus as snx
+
+from ess.reduce.nexus.workflow import GenericNeXusWorkflow
+from ess.spectroscopy.types import (
+    GoniometerAngles,
+    NeXusClass,
+    NeXusFileSpec,
+    RunType,
+)
+
+
+# See https://github.com/scipp/essreduce/issues/98
+def moderator_class_for_source() -> NeXusClass[snx.NXsource]:
+    """Select NXmoderator as the source."""
+    return NeXusClass[snx.NXsource](snx.NXmoderator)
+
+
+def load_goniometer_angles(
+    file_spec: NeXusFileSpec[RunType],
+) -> GoniometerAngles[RunType]:
+    # TODO need mechanism in ESSreduce to load specific components of non-unique
+    #  class by name
```

Comment:
You mean to load multiple datasets from `NXparameters`? You could use an application definition for filtering. But the below is also fine.

Reply:
How would I use an application definition? This code may use an already open file, so we can't add new definitions here. So this would have to be done when first opening the file. And would this not interfere with loading other logs?